### ISSUE TITLE
dynamic is not assignable to bool

### ISSUE THREAD

**Issue by zoechi** [dart-lang/angular_analyzer_plugin#523]
![image](https://user-images.githubusercontent.com/405837/37761142-23551f9c-2db9-11e8-915c-c43e5a165854.png)
```dart
  Stream<bool> get isLoggedIn =>
      authenticationController.onAuthenticatedUserChange
          .map((user) => user != null);
```

**Comment by zoechi** [comment]
Similar
![image](https://user-images.githubusercontent.com/405837/37761227-68f12dd4-2db9-11e8-86fe-ab089ba39f04.png)

```
Stream<BuiltList<ImageReferenceController>> get onImageReferences =>
      _onImageReferences ??=
          _imageReferenceEditorController.onImageReferencesChange;
```
```
class ImageReferenceController {
  bool get isSelected =>
      _imageReference.sys.id ==
      store.state.backoffice.imageReferenceEditor.selectedImageReference?.sys
          ?.id;
```

**Comment by MichaelRFairhurst** [comment]
Thanks for filing, @zoechi. This is very curious and I'll start digging.

**Comment by MichaelRFairhurst** [comment]
Rereading more closely, I see these all use the async pipe. We don't fully resolve pipes yet, we just say the result is dynamic. This has been a want-to-solve on my list for a while, it will be fun :)

However, we still shouldn't be seeing assignment errors in this case. Even if it were dynamic, it'd be assignable.

I recently made some changes to the type system to leverage strong mode...do you have strong mode on or off? I may be getting the "dynamic" type incorrectly for one case or the other.

**Comment by MichaelRFairhurst** [comment]
related to #121

**Comment by MichaelRFairhurst** [comment]
Ah yes. We do support

```
<!-- @ngIgnoreErrors: foo, bar, baz -->
```

at the top of the file currently

**Comment by zoechi** [comment]
@MichaelRFairhurst thanks for the reminder.
I forgot aboit that syntax-

**Comment by MichaelRFairhurst** [comment]
Thisg should be fixed as pipes are now supported in #621
